examples/manifold/plot_compare_isomaps.py

```diff
@@ -0,0 +1,83 @@
+"""
+=========================================
+ Comparison of Isomap methods
+=========================================
+
+An illustration of dimensionality reduction on the Swiss-roll
+data set with both Isomap and Landmark Isomap (L-Isomap) methods.
+"""
+
+# Author: Lucas David -- <lucasolivdavid@gmail.com>
+# License: BSD 3 clause, (C) 2015
+
+print(__doc__)
+
+from time import time
+import matplotlib.pyplot as plt
+from matplotlib.ticker import NullFormatter
+from mpl_toolkits.mplot3d import Axes3D
+from sklearn import manifold, datasets
+
+# Next line to silence pyflakes. This import is needed.
+Axes3D
+
+isomap_params = {'n_neighbors': 10, 'n_components': 2}
+l_isomap_params = {'n_neighbors': 10, 'n_components': 2, 'landmarks': 'auto'}
+
+comparisons = [
+    (
+        n_samples,
+        (manifold.Isomap, isomap_params),
+        (manifold.Isomap, l_isomap_params),
+    ) for n_samples in range(1000, 16001, 5000)]
```

Comment:
The idea here was to show how the difference between both algorithms' performances increases as the number of samples increases, and how L-Isomap can handle this high number of samples whereas Isomap cannot. To avoid memory collapse the fitting happens inside a [try/raise](https://github.com/scikit-learn/scikit-learn/pull/5969/files#diff-a01c99fa301b88ee5a62d79240cfe3e8R78) clause:
```py
try:
    t0, t1 = time(), None
    ...
except MemoryError as e:
    print('Failed.' % j)
```

This should be protecting the build against memory errors, shouldn't it? Anyhow, I'll reduce and see if it works.